File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        ans = []
        prod = 1
        for i in range(len(nums)):
            for j in range(len(nums)):
                if i == j:
                    continue
                prod *= nums[j]

            ans.append(prod)
            prod = 1

        return ans

        
        

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    static bool cmp(const pair<int, int> &a, const pair<int ,int> &b){
        return a.second > b.second;
    }

    vector<int> topKFrequent(vector<int>& nums, int k) {
        map<int, int> frequencyMap;

        for (auto num : nums) {
            frequencyMap[num]++;
        }

        vector<pair<int, int>> frequencyVector;

        // Fill frequencyVector with the entries from frequencyMap
        for (auto& entry : frequencyMap) {
            frequencyVector.push_back(entry);
        }

        // Sort by frequency in descending order
        sort(frequencyVector.begin(), frequencyVector.end(), cmp);

        vector<int> ans;
        for (int i = 0; i < k; i++) {
            ans.push_back(frequencyVector[i].first);  
        }

        return ans;
    }
};
